conformance/data_model/annotations.ion

```diff
@@ -0,0 +1,48 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO:
+// * Ion 1.0 binary annotations
+// * In Ion 1.1 binary,
+//   * SID annotations
+//   * FlexSym annotations (inline, SID, and system symbols)
+// * In text (any version)
+//   * Text Annotations (quoted and unquoted)
+//   * Text SID Annotations
+//   * non-symbols cannot be used as annotations
+//   * specifically, annotations must be single quoted if they are quoted
+//   * operator symbols must be quoted to be used as an annotation, even in s-expressions
+//   * dangling annotations must signal an error
+// * In binary (any version)
+//   * Annotations cannot be applied to NOP
+// * In all encodings
+//   * Any number of annotations is allowed (in any encoding)
+//   * Annotations can be applied to any type of value (all prior tests should be annotating a small integer)
+//
+// Less common scenarios:
+// * Add a test case with multiple values that are annotated to help ensure that there's
+//   no leaking annotation state between values
+
+(ion_1_x "in text, unquoted keywords cannot be used as an annotation"
```

Comment:
Maybe add a `TODO` for the inverse of this? (i.e. Quoted keywords can be used as an annotation.)